antarest/storage/repository/filesystem/matrix/date_serializer.py

```diff
@@ -0,0 +1,205 @@
+import re
+from abc import ABC, abstractmethod
+from typing import Tuple, Any, List
+
+import pandas as pd  # type: ignore
+
+
+class IDateMatrixSerializer(ABC):
+    _MONTHS = {
+        "JAN": "01",
+        "FEB": "02",
+        "MAR": "03",
+        "APR": "04",
+        "MAY": "05",
+        "JUNE": "06",
+        "JULY": "07",
+        "AUG": "08",
+        "SEPT": "09",
+        "OCT": "10",
+        "NOV": "11",
+        "DEC": "12",
+    }
+    _R_MONTHS = {v: k for k, v in _MONTHS.items()}
+
+    def __init__(self, area: str):
+        self.area = area
+
+    @abstractmethod
+    def extract_date(self, df: pd.DataFrame) -> Tuple[pd.Index, pd.DataFrame]:
+        raise NotImplementedError()
+
+    @abstractmethod
+    def build_date(self, index: pd.Index) -> pd.DataFrame:
+        raise NotImplementedError()
+
+
+class HourlyMatrixSerializer(IDateMatrixSerializer):
+    def build_date(self, index: pd.Index) -> pd.DataFrame:
+        def _map(row: str) -> Tuple[str, int, str, str, str]:
+            h, d, m = re.split("[\s/]", row)
+            return "", 1, d, IDateMatrixSerializer._R_MONTHS[m], h
+
+        items = index.map(_map).tolist()
+        matrix = pd.DataFrame(items)
+        matrix[1] = matrix[1].cumsum()
+
+        headers = pd.DataFrame(
+            [
+                [self.area.upper(), "hourly", "", "", ""],
+                ["", "", "", "", ""],
+                ["", "index", "day", "month", "hourly"],
+            ]
+        )
+
+        return pd.concat([headers, matrix], axis=0)
+
+    def extract_date(self, df: pd.DataFrame) -> Tuple[pd.Index, pd.DataFrame]:
+        # Extract left part with date
+        date = df.loc[2:, ["Unnamed: 2", "Unnamed: 3", "Unnamed: 4"]]
+        date.columns = ["day", "month", "hour"]
+        date["month"] = date["month"].map(IDateMatrixSerializer._MONTHS)
+        date = (
+            date["hour"] + " " + date["day"] + "/" + date["month"].astype(str)
```

Comment:
when using this as index, the sorting is not correct (as seen in the test resource file de_fr_values_hourly).
at least, switch the order (month > day > hour).
Yet there is another potential issue because sometime the series does not begin with the january (can start on july and end with june..)

Reply:
the last issue, is real. There is (and it's the case most often than not) cases when the first date is something like "01	JUL	00:00" and the last "29	JUN	23:00".
So the index should be the very first column (integer from 1 to 8760 in a hourly scale) and add a new column containing the date

Reply:
This was a tricky issue. Since we serialize Dataframe in dict with index as key, we don't have any control of order. I change the dict to format to `split` ([docs)](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.to_dict.html)), there are many improvment with this format: 
- index is a list, No issue any more with random order
- columns is a list, we can use tuple as column whereas before we can do only string column
- data size decrease.